workers/summarizer.py

```diff
@@ -22,38 +24,60 @@
 llm, embeddings = load_model()
 
 
+# even though a single task takes a long time to complete,
+# as soon as one task is started, all elements of the queue are released
+task_queue = []
+task_queue_limit = 10
+
+
+def extract_uuid(task):
+    return task["uuid"]
+
+
 def summarize():
+    global task_queue
 
-    task = db_get_incomplete_completion_task()
+    queue_space = task_queue_limit - len(task_queue)
+    task_queue += db_get_incomplete_completion_tasks(queue_space)
+    current_task = task_queue[0]
 
-    if task is None:
-        return
+    # find the first task ready for execution, dismiss the others
+    for task in task_queue:
+        # check all dependencies for completeness
+        dep_list = task["required_crawl_tasks"]
 
-    def get_query():
-        return WebQuery(task["mode"].lower(), prompt_core=task["prompt"])
+        if db_are_tasks_completed(dep_list):
```

Comment:
This is a mistake, i believe i should have used a list-accepting variation of `db_is_crawl_task_fully_embedded`,
which actually checks if the task is ready to be used.
```suggestion
        if db_are_crawl_tasks_fully_embedded(dep_list):
```